practice-two/src/components/Toast/Toast.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useRef } from "react";
+import "./Toast.css";
+import PropTypes from "prop-types";
+
+const Toast = ({ message, backgroundColor }) => {
+  const toastContainerRef = useRef(null);
+
+  useEffect(() => {
+    if (toastContainerRef.current) {
+      const container = toastContainerRef.current;
+
+      setTimeout(() => {
+        container.classList.add("toast__close");
+      }, 4000);
+    }
+  }, []);
+
+  return (
+    <div
+      style={{ backgroundColor: backgroundColor }}
+      ref={toastContainerRef}
+      className={`toast__container`}
+    >
+      <p className='toast__message'>{message}</p>
+    </div>
+  );
+};
+
+Toast.propTypes = {
+  message: PropTypes.string.isRequired,
+  backgroundColor: PropTypes.string.isRequired,
+};
+
+Toast.defaultProps = {
+  message: "Success",
+  backgroundColor: "#5cb85c",
+};
+
+export default Toast;
```

Comment:
how about check re-render component?

Reply:
Currently, I have successfully checked that when the API call fails, it will display a message on the screen as follows
![image](https://github.com/datdragon777/React-Training/assets/74806826/1b0b021e-dc40-4d75-9f03-a378417fb907)

Reply:
my mean is check re-render component not render
It's mean performance re-render

Reply:
I am using react dev tool, it has Profiler to check performance of each performance of each component.
For example, I checked performance of Toast component, it's green which means it is having the best re-render performance
![image](https://github.com/datdragon777/React-Training/assets/74806826/1562c706-06f2-4c77-a476-7897a46a1403)